subsys/bluetooth/controller/hal/nrf5/radio.c

```diff
@@ -798,15 +812,22 @@ void *radio_ccm_rx_pkt_set(struct ccm *ccm, void *pkt)
 
 void *radio_ccm_tx_pkt_set(struct ccm *ccm, void *pkt)
 {
+	u32_t mode;
+
 	NRF_CCM->ENABLE = CCM_ENABLE_ENABLE_Disabled;
 	NRF_CCM->ENABLE = CCM_ENABLE_ENABLE_Enabled;
-	NRF_CCM->MODE =
-#if !defined(CONFIG_SOC_SERIES_NRF51X)
-	    ((CCM_MODE_LENGTH_Extended << CCM_MODE_LENGTH_Pos) &
-	       CCM_MODE_LENGTH_Msk) |
+	mode = (CCM_MODE_MODE_Encryption << CCM_MODE_MODE_Pos) &
+	       CCM_MODE_MODE_Msk;
+#if defined(CONFIG_SOC_SERIES_NRF52X)
+	mode |= (CCM_MODE_LENGTH_Extended << CCM_MODE_LENGTH_Pos) &
```

Comment:
Can you comment on why the CCM_MODE_LENGTH_Extended is required here regardless of actual current PHY active?

Reply:
Sure.